src/ark/segmentation/fiber_segmentation.py

```diff
@@ -294,9 +353,147 @@ def segment_fibers(data_xr, fiber_channel, out_dir, fov, blur=2, contrast_scalin
 
     if save_csv:
         fiber_object_table.to_csv(os.path.join(out_dir, 'fiber_object_table.csv'))
 
     return fiber_object_table
+
+
+def calculate_density(fov_fiber_table, total_pixels):
+    """ Calculates both pixel area and fiber number based densities.
+    pixel based = fiber pixel area / total image area
+    fiber number based = number of fibers / total image area
+
+    Args:
+        fov_fiber_table (pd.DataFrame):
+            the array representation of the fiber segmented mask for an image
+        total_pixels (int):
+            area of the image
+
+    Returns:
+        tuple (float, float):
+         - returns the both densities scaled up by 100
+    """
+
+    fiber_num = len(np.unique(fov_fiber_table.label))
+    fiber_density = fiber_num / total_pixels
+
+    pixel_sum = np.sum(fov_fiber_table['area'].values)
+    pixel_density = pixel_sum / total_pixels
+
+    return pixel_density * 100, fiber_density * 100
+
+
+def generate_summary_stats(fiber_object_table, fibseg_dir, tile_length=512, save_tiles=False):
+    """ Calculates the fov level and tile level statistics for alignment, length, and density.
+    Saves them to separate csvs.
+
+    Args:
+        fiber_object_table (pd.DataFrame):
+            dataframe containing the fiber objects and their properties (fov, label, alignment,
+            centroid-0, centroid-1, major_axis_length, minor_axis_length)
+        fibseg_dir (string):
+            path to directory containing the fiber segmentation masks
+        tile_length (int):
+            length of tile size, must be a factor of the total image size (default to 512)
+        save_tiles (bool):
+            whether to save cropped images (default to False)
+
+    Returns:
+        tuple (pd.DataFrame, pd.DataFrame):
+         - returns the both fov and tile stats
+    """
+
+    io_utils.validate_paths(fibseg_dir)
+    # this makes sure tile length is a factor of 1024 and 2048
+    if 1024 % tile_length != 0:
+        raise ValueError("Tile length must be a factor of the minimum image size.")
+
+    save_dir = os.path.join(fibseg_dir, f'tile_stats_{tile_length}')
+    fovs = np.unique(fiber_object_table.fov)
+    fov_stats, tile_stats = [], []
+    fov_alignment, fov_avg_length, fov_pixel_density, fov_fiber_density = [], [], [], []
+
+    # get fov level and tile level stats for each image
+    for fov in fovs:
+        fov_fiber_img = io.imread(os.path.join(fibseg_dir, fov + '_fiber_labels.tiff'))
+        fov_length = fov_fiber_img.shape[0]
+        fov_table = fiber_object_table[fiber_object_table.fov == fov]
+
+        # fov level stats
+        fov_align_scores = fov_table['alignment_score'].values
+        fov_alignment.append(np.mean(fov_align_scores[~np.isnan(fov_align_scores)]))
+        fov_avg_length.append(np.mean(fov_table['major_axis_length'].values))
+        fov_p_density, fov_p_density = calculate_density(fov_table, fov_length**2)
```

Comment:
Let's use different variable names for these two: maybe `fov_pixel_density` and `fov_fiber_density`.

Reply:
I use those names up above for the lists. 🙃 That's why I ended up using the abbreviated versions for the values lol